### PR TITLE
Add upload ID photo to committee

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Committee/CommitteeManager.php [hunk 1]
```diff
@@ -27,10 +30,14 @@ class CommitteeManager
     private const COMMITTEE_PROPOSALS_COUNT = 3;
 
     private $registry;
+    private $storage;
```
Comment (HeahDude) [inline src/Committee/CommitteeManager.php]: Please don't add new dependencies for one method, create a new service instead. Like `PhotoManager`.

File: src/Committee/CommitteeCreationCommandHandler.php [hunk 2]
```diff
@@ -14,23 +14,30 @@ class CommitteeCreationCommandHandler
...
         $this->mailer = $mailer;
+        $this->committeeManager = $committeeManager;
     }
 
     public function handle(CommitteeCreationCommand $command): void
     {
         $adherent = $command->getAdherent();
         $committee = $this->factory->createFromCommitteeCreationCommand($command);
+        // Uploads an ID photo
+        if (null !== $command->getPhoto()) {
```
Comment (HeahDude) [inline src/Committee/CommitteeCreationCommandHandler.php]: Use `hasPhoto` or store the result on the first call like `null !== $photo = $command->getPhoto()`.
I suggest to either do that in the manager, or to pass it the committee and the photo.
The command must protect the "real" committee to be in an invalid state.

File: src/Committee/CommitteeUpdateCommandHandler.php [hunk 3]
```diff
@@ -12,15 +12,18 @@ class CommitteeUpdateCommandHandler
...
     private $addressFactory;
     private $manager;
+    private $committeeManager;
 
     public function __construct(
         EventDispatcherInterface $dispatcher,
         ObjectManager $manager,
-        PostAddressFactory $addressFactory
+        PostAddressFactory $addressFactory,
+        CommitteeManager $committeeManager
```
Comment (HeahDude) [inline src/Committee/CommitteeUpdateCommandHandler.php]: Same here don't inject the whole manager, only the photo uploader service.

File: src/Entity/Committee.php [hunk 4]
```diff
@@ -142,6 +163,41 @@ public function getPostAddress(): PostAddress
...
+    {
+        $this->urlPhoto = $url;
+    }
+
+    public function getUrlPhoto(): string
+    {
+        return $this->urlPhoto;
+    }
+
+    public function getPhoto(): ?UploadedFile
```
Comment (HeahDude) [inline src/Entity/Committee.php]: The upload should live in the command, not the entity.

File: src/Form/CommitteeCommandType.php [hunk 5]
```diff
@@ -41,6 +44,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'default_protocol' => null,
             ])
         ;
+
+        if (!$committee || $committee->isWaitingForApproval()) {
+            $builder->add('photo', FileType::class, [
```
Comment (HeahDude) [inline src/Form/CommitteeCommandType.php]: If a user submit this filed as empty, will it erase the photo?

File: templates/adherent/create_committee.html.twig [hunk 6]
```diff
@@ -30,7 +30,7 @@
             </header>
 
             <div class="committee__form">
-                <form method="post" id="create-committee-form" class="form committee-form">
+                {{ form_start(form, {attr: {'id': 'create-committee-form', 'class': 'form committee-form'}}) }}
```
Comment (HeahDude) [inline templates/adherent/create_committee.html.twig]: CS

File: templates/committee/_form.html.twig [hunk 7]
```diff
@@ -57,6 +57,20 @@
 
 </div>
 
+{% set committee = committee is defined ? committee : form.vars.data.committee %}
+{% if not committee or committee.isWaitingForApproval %}
+    <div class="form__row" id="field-photo">
+        {{ form_label(form.photo, "Ajouter votre photo d'identité", {'label_attr': {'class': 'text--bold'}}) }}<br />
+        {% if committee and committee.photoUploaded %}
```
Comment (HeahDude) [inline templates/committee/_form.html.twig]: `committee.hasPhotoUploaded`

File: tests/Committee/CommitteeManagerTest.php [hunk 8]
```diff
@@ -267,7 +267,7 @@ protected function setUp()
         ]);
 
         $this->container = $this->getContainer();
-        $this->committeeManager = new CommitteeManager($this->getManagerRegistry());
+        $this->committeeManager = new CommitteeManager($this->getManagerRegistry(), $this->getStorage(), $this->getGlide());
```
Comment (HeahDude) [inline tests/Committee/CommitteeManagerTest.php]: To revert.

File: src/Form/CommitteeCommandType.php (hunk 5 above)
Comment (illusionOfParadise) [reply src/Form/CommitteeCommandType.php]: No, we have not remove the uploaded file only replace with new one.

File: src/Committee/PhotoManager.php [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+class PhotoManager
+{
+    private $registry;
+    private $storage;
+    private $glide;
+
+    public function __construct(ManagerRegistry $registry, Filesystem $storage, Server $glide)
+    {
+        $this->registry = $registry;
```
Comment (HeahDude) [inline src/Committee/PhotoManager.php]: Seems unused.

File: src/Committee/PhotoManager.php [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * Uploads and saves the ID photo of committee creator.
+     *
+     * @param Committee $committee
+     */
+    public function addPhotoFromCommand(CommitteeCommand $command, Committee $committee): void
+    {
+        if (null !== $photo = $command->getPhoto()) {
+            if (!$photo instanceof UploadedFile) {
+                throw new \RuntimeException(sprintf('The photo must be an instance of %s', UploadedFile::class));
```
Comment (HeahDude) [inline src/Committee/PhotoManager.php]: Can be removed though.

File: src/Committee/PhotoManager.php [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+    /**
+     * Uploads and saves the ID photo of committee creator.
+     *
+     * @param Committee $committee
+     */
+    public function addPhotoFromCommand(CommitteeCommand $command, Committee $committee): void
+    {
+        if (null !== $photo = $command->getPhoto()) {
+            if (!$photo instanceof UploadedFile) {
```
Comment (HeahDude) [inline src/Committee/PhotoManager.php]: This is the same check as just before.

File: src/Entity/Committee.php [hunk 12]
```diff
@@ -91,6 +98,18 @@ class Committee extends BaseGroup implements CoordinatorAreaInterface
...
+     * @var bool
+     *
+     * @ORM\Column(type="boolean", options={"default"=false})
+     */
+    private $photoUploaded = false;
+
+    /**
+     * @var UploadedFile|null
+     */
+    private $photo;
```
Comment (HeahDude) [inline src/Entity/Committee.php]: Unused, can be removed.

File: src/Entity/Committee.php [hunk 13]
```diff
@@ -277,8 +316,12 @@ public function getSocialNetworksLinks(): array
         return $links;
     }
 
-    public function update(string $name, string $description, PostAddress $address, PhoneNumber $phone): void
```
Comment (HeahDude) [inline src/Entity/Committee.php]: Why this change?

File: src/Form/CommitteeCommandType.php [hunk 14]
```diff
@@ -15,6 +17,7 @@ class CommitteeCommandType extends AbstractType
 {
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        /** @var Committee $committee */
         $committee = $builder->getData() instanceof CommitteeCommand ? $builder->getData()->getCommittee() : null;
```
Comment (HeahDude) [inline src/Form/CommitteeCommandType.php]: Please use a var for `$builder->getData()` and remove the phpdoc.

File: src/Committee/CommitteeCommand.php [hunk 15]
```diff
@@ -63,6 +64,16 @@ class CommitteeCommand
      */
     public $googlePlusPageUrl;
 
+    /**
+     * @var UploadedFile|null
+     *
+     * @Assert\Image(
+     *     maxSize = "50M",
```
Comment (HeahDude) [inline src/Committee/CommitteeCommand.php]: Maybe you intended to put `5M` here?

File: templates/committee/_form.html.twig [hunk 16]
```diff
@@ -57,6 +57,20 @@
 
 </div>
 
+{% set committee = committee is defined ? committee : form.vars.data.committee %}
+{% if not committee or committee.isWaitingForApproval %}
+    <div class="form__row" id="field-photo">
+        {{ form_label(form.photo, "Ajouter votre photo d'identité", {'label_attr': {'class': 'text--bold'}}) }}<br />
```
Comment (Remg) [inline templates/committee/_form.html.twig]: useless quotes for keys

File: templates/committee/_form.html.twig [hunk 17]
```diff
@@ -57,6 +57,20 @@
...
+        {% if committee and committee.hasPhotoUploaded %}
+            <img style="max-width: 100px" src="{{ path('asset_url', { path: committee.photoPath }) }}" />
+        {% endif %}
+        {{ form_errors(form.photo) }}
+        {{ form_widget(form.photo, { 'attr': {
+            'class': 'form--full',
```
Comment (Remg) [inline templates/committee/_form.html.twig]: same here
